### DISCUSSION TITLE
Training and predicting on concated datasets

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#5892] [General]
So i create the big dataset, then split it into separate ones and zeros to create a "known" set to simulate the data that has the ground truth information, and the "unknown" set that does not.

From the known set, I create 2 sets, ones and zeros: 
```
ones = known.copy(idx=range(split))
zeros = known.copy(idx=range(split))
```
Then build balanced training and validation from that concat it using torch.
```
N = 10000
known_ones = ones.copy(idx=range(int(N*split)))
known_zeros = zeros.copy(idx=range(N-len(known_ones)))

known_dataset = torch.utils.data.ConcatDataset([known_ones, known_zeros])

unknown_ones = ones.copy(idx=range(int(N*split), len(ones)))
unknown_zeros = zeros.copy(idx=range(N-len(known_ones), len(zeros)))

unknown_dataset = torch.utils.data.ConcatDataset([unknown_ones, unknown_zeros])

from torch_geometric.data import DataLoader

train_loader = DataLoader(known_dataset, batch_size=64, shuffle=False)
```

I train, on the known, and predict the unknown. The predictions all produce zero! 
Now in past experiments where I just training on the dataset by creating the training, val, and test from that, and achieve very good accuracy scores. But as soon as I started doing it this way with concat, its no longer working. Any ideas? If you need to see more code let me know.

**Comment by rusty1s** [comment]
You might want to run with `shuffle=True`. Otherwise, the last steps in your training only train against zero labels.
